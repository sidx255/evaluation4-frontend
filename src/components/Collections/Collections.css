
.collections {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
    padding-left: 20px; 
}

.content-type {
    display: flex;
    width: 280px;
    height: 80px;
    margin: 0px;
    padding-left: 20px;
    background-color: #2f2f2f;
    color: #ffffff;
    font-size: 15px;
    font-weight: 600;
    align-items: center;
    justify-content: flex-start;
}

/* .collection {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 300px;
    height: 300px;
    margin: 10px;
    background-color: #ffffff;
    border-radius: 10px;
    box-shadow: 0px 0px 10px 0px rgba(0,0,0,0.75);
} */