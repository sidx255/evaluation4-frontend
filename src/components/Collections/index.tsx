
import React, { useEffect } from 'react';
// import './index.css';

import './Collections.css';

import makeRequest from '../../utils/makeRequest';
import {GET_COLLECTIONS} from '../../constants/apiEndPoints';

// make a request to the backend to get the collections
// and then render them in the sidebar


const Collections = ({setCollection}:any) => {
  const [collections, setCollections] = React.useState([]);
  useEffect(() => {
    makeRequest(GET_COLLECTIONS, {
      headers: {
        'Content-Type': 'application/json',
        'authorization': `${localStorage.getItem('token')}`
      }
    })
      .then((res: any) => {
        setCollections(res);
        // console.log(collections);
      }).catch((err: any) => {
        console.log(err);
      });

  }, []);
    
  return (
    <div>
      <div className='collections'>
        {collections.map((collection: any, idx:any) => {
          return (
            <div key={collection.id} className="collections-text">
              <p onClick={()=> setCollection(collection)}>* {collection.collection_name}</p>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default Collections;
    