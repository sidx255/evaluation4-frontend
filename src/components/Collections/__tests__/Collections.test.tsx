
// snapshot test for Collections component

import React from 'react';
import { render } from '@testing-library/react';

import Collections from '../index';

describe('Collections', () => {
  it('should render correctly', () => {
    const { container } = render(<Collections />);
    expect(container).toMatchSnapshot();
  });
});