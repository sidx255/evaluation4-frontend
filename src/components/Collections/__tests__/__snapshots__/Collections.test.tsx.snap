// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections should render correctly 1`] = `
<div>
  <div>
    <div
      class="collections"
    />
  </div>
</div>
`;
